spec/lib/index.spec.js

```diff
@@ -21,15 +21,74 @@ describe("lib/index", () => {
 
       // Invalid options.entry
       expect(() => { new Interlock(_.merge({}, minimalValidConfig, { entry: true })); })
-        .to.throw(Error);
+        .to.throw(Error, "child \"entry\" fails because [\"entry\" must be an object]");
```

Comment:
I found out the hard way that `expect.to.throw` basically just swallows all errors which means that errors caused by bad formatting or an invalid Joi schema will cause a false positive. There were two ways to avoid this:

Ideally I'd use the Chai [throw](http://chaijs.com/api/bdd/#-throw-constructor-) with an Error constructor as a predicate but Joi doesn't [do custom Errors properly](https://github.com/hapijs/joi/search?utf8=%E2%9C%93&q=ValidationError) so this was not an option. The next best thing seemed to pass the Joi error string to `throw` but it made the code quite a mess. You can (as Joi does) just check that `err.name === "ValidationError"` but that would involve removing the .throw and wrapping everything in a try/catch. I'm happy to convert to something else but I thought I'd comment as to why this code looks like hell :P.